### DISCUSSION TITLE
Conditional keys in ObjectNode?

### DISCUSSION THREAD

**Discussion by blockninja124** [misode/misode.github.io#625] [Q&A]
I saw the modded generators part of the website, and thought that if I could make a generator for my mods datapack support it might get added through a pull request (or not, but no harm trying). Anyway, in the process of trying to make a generator, I found it quite tricky to have conditional keys? Such as if weather set to true, show weather color option. I found in the immersive weathering generator that a Switch and Case are used, but that appears to only be for enums? And also only lets me use it once per ObjectNode? Anyway, I was wondering if theres an easier way. Perhaps using the SwitchNode, although I couldn't figure out how to use it. Any info on how to do this would be appreciated!

**Comment by misode** [comment]
Using the mcschema library, you can use the `Mod` node with the `enabled` option, but it's faily hacky. You can see a recent use of that here: https://github.com/misode/minecraft-schemas/commit/9440e2cb5472ccd7c1390a84714e25b51b95dc17

Though I have to warn you, I am in the process of replacing the mcschema system with something new, progress: https://github.com/misode/misode.github.io/pull/614. This will make it much much easier to write custom generators. If you're willing to wait a month or so for me to finish that PR.

**Reply by blockninja124** [reply]
Interesting. I guess I'll use the hacky system for now, but since I probably won't spend enough time on this to get it ready to PR before your improvement, I'll probably switch it to that system eventually also. Thanks!
